src/legacy/core_plugins/kibana/public/dashboard/np_ready/application.ts

```diff
@@ -67,21 +69,46 @@ export interface RenderDeps {
   embeddables: IEmbeddableStart;
   localStorage: Storage;
   share: SharePluginStart;
+  kbnUrlStateStorage: IKbnUrlStateStorage;
+  history: History;
+
+  // hack: keeping the value nested
+  // as configureAppAngularModule() is called once and initialised with reference to 'hasInheritedGlobalState' ,
+  // so to be able to update underlying value, so angular controller could pick it up
+  // this param is used to determine if time filter saved with dashboard should be applied or not
+  hasInheritedGlobalState?: {
+    value: boolean;
+  };
 }
 
 let angularModuleInstance: IModule | null = null;
+const hasInheritedGlobalStateRef: { value: boolean } = { value: false };
 
 export const renderApp = (element: HTMLElement, appBasePath: string, deps: RenderDeps) => {
+  const { stop: stopSyncingGlobalFilters, hasInheritedFiltersFromUrl } = syncFilters(
+    deps.kbnUrlStateStorage,
+    deps.npDataStart.query.filterManager,
```

Comment:
I'd pass `npDataStart.query` in

Reply:
Improved 👍

Reply:
Also as agreed, renamed to syncQuery (not filters)